### DISCUSSION TITLE
Fatigue Analysis of OC4 JACKET

### DISCUSSION THREAD

**Discussion by Krishnaraj18** [OpenFAST/openfast#1749] [Q&A]
I have a doubt regarding OC4 jacket simulation in openfast.How can i do fatigue analysis of OC4 jacket in openfast?should i get the stress concentration factor of each joint of the jacket? or not?

**Comment by jjonkman** [comment]
Dear @Krishnaraj18,

I'm not sure I fully understand your question.  Certainly in OpenFAST you can calculate the member-level loads (sectional forces and bending moments) in members of jacket support structures for a range of conditions (load cases), including for the NREL 5-MW baseline wind turbine atop the OC4 jacket.  However, OpenFAST does not compute stress or strain directly, and as such, does not directly consider stress concentration factors.

Perhaps the following OpenFAST discussion is of use to you: https://github.com/OpenFAST/openfast/discussions/1747?

Best regards,

**Reply by Krishnaraj18** [reply]
Sir, I have one doubt regarding fatigue analysis of OC3 wind turbine. How can I do fatigue analysis using m life method in OpenFast. Can I get the code of m-life input file.

**Reply by Krishnaraj18** [reply]
Fatigue can be done using rainflow counting method also right.Is there any chance of getting the code for that for OC3 monopile wind turbines.

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

MLife performs fatigue post-processing based on time series generated by OpenFAST, including rainflow cycle counting.  MLife is available here and includes several examples in the CertTest that you can use as guidance: https://www.nrel.gov/wind/nwtc/mlife.html.

Best regards,

**Reply by Krishnaraj18** [reply]
What are the input parameter I have to change for OC4 jacket wind turbine in operating condition and extreme shutdown condition?

**Reply by Krishnaraj18** [reply]
And how can i calculate the natural frequencies of whole jacket supported wind turbine(whole structure)?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

The CertTest example for the NREL 5-MW baseline wind turbine atop the OC4-jacket is already set up for normal operating conditions; of course, you can change the wind conditions (and associated initial conditions) or sea state if you want to run different cases.

By "extreme shutdown condition", do you mean the emergency stop load case (DLC 5.1 from the IEC 61400-3-1 standard?  I'm not aware that an emergency stop protocol has been defined for the NREL 5-MW baseline wind turbine; you'll likely have to make your own assumptions for your own purposes.

You can compute full-system natural frequencies (including the jacket substructure) using an OpenFAST linearization analysis, followed by eigenenanalysis.

Best regards,

**Reply by Krishnaraj18** [reply]
Dear Jjonkman,
I have some doubt regarding Pile soil interaction of NREL OC4 jacket wind turbine. I want to know that if I give SSI file in SubDyn what about the response change in blade deflection, tower base moment and other responses?Becaude I didn't noticed any change in response while i am giving SSI file. I am confused because the SSI have great effect on the structural response right, it will reduce the eigen frequency right? So response also will change right? Can you please help me to clarify this.

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

Yes, I would expect that introducing soil stiffness in SubDyn through the `SSIfile` would impact the system response.  If you are not seeing any change, when you set the `SSIfile`, are you unlocking the relevant reaction joint translation/rotation degrees of freedom (`RctTDXss`, `RctTDYss`, ... `RctRDZss` = 0)  where you have placed the stiffness?

Best regards,

**Reply by Krishnaraj18** [reply]
Ok i will do it and check regarding this. I have one more doubt i run the simulation by varying wind speed from 1 to 50 m/s and edited corresponding Hs and Tp that i derived using weibul fit from the location Gulf of Mannar.I did the simulation like 1 to 25 operating condition and 26 to 50 parked mode.The thing is i am getting blade deflection maximum at 11.4m/s then it decreases.In case of tower i am getting an increasing curve from 1 to 11.4 rang and then decreasing after 25 m/s tower top deflection increasing.I want to know that this pattern of tower top deflection  is correct or not.If the deflection in blade increases how it effect on the tower top deflection and it's base moment,it will also increase or not?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

I'm not sure I fully understand your question.  It would help if you shared more information on your model set and results (including plots of mean and max blade and tower deflections as a function of mean wind speed).

Regardless, overall your description sounds reasonable because I would expect that the maximum mean rotor thrust to occur at rated wind speed (11.4 m/s for the NREL 5-MW baseline wind turbine) because the blades with pitch and shed thrust at higher mean wind speeds.

Best regards,

**Reply by Krishnaraj18** [reply]
I am attaching  the maximum responses like outplane blade deflection(OPD),Inplane(IPD),Tower top displacement(TTDFA-fore-aft and TTD(SS)side to side),RootMoment(RootMx,RootMy,RootMz),Mudline moment(BMx,BMy,BMz).I drawan the maximum tower top deflection(Fore-aft) for wind speed 1-50,here 1 to 25m/s is Normal operating mode,26 to 50 is parked mode.This showing an increase in curve then decreases in the same range of 1 to 25 m/s.This is correct or not?the blade maximum deflection at 11.4 m/s will effect this tower top deflection ?what about the other responses it also depends on the blade deflection behavior?


[Maximum response-github.xlsx](https://github.com/OpenFAST/openfast/files/14349619/Maximum.response-github.xlsx)

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

Your results generally make sense to me.  Again, the maximum at rated wind speed occurs as a result of the variable speed and variable pitch controller.  Once the rotor is stopped, and the blades are feathered for the idling condition, the thrust will drop a lot, but slowly increase as the wind speed increases.  This is what your results show.

Best regards,

**Reply by Krishnaraj18** [reply]
This blade deflection will effects tower top deflection. I am getting same trend in case of tower top also. I have one more doubt. In operating mode I am getting out-plane deflection more than in-plane deflection but in parked mode I am getting in-plane  blade deflection higher than out-plane deflection, approximately 10 times more than out-plane. Is it correct or not?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

That makes sense to me too.  In operating conditions, the flapwise mode (which is much more flexible than the edgewise mode) dominates the out-of-plane deflection.  But in the parked condition (with the blades feathered with 90degrees pitch), the flapwise mode would dominate the in-plane deflection.

Best regards,

**Reply by Krishnaraj18** [reply]
The deflection in blade will effects the tower also right? Maximum response in tower and jacket structure(Support structure depends on the blade deflection or not? I referred one journal ,in that journal it showing 2 m blade deflection at 25 m/s wind speed in operating mode and 1.5 m deflection in parked mode at same wind speed(25m/s).But for me is like 2 and 0.2m respectively.And if i am changing normal operating mode to parked mode, I changed wakemod to 0 , AFAeroMod 1,TPCon to 9999.9,blade pitch 90 degree.Actually in high wind comes the pitch angle will change right ,in practical how it stand on 90 degree.Because 90 degree means the effects on the blade is minimum right.but the deflection will be high in high wind speed right? I n parked mode what about the override pitch manoeuvring value?If i am analysing the structure under parked condition from cut in to cut out wind speed,what is the pitch angle 0 or 90 degree.?I did it for 90 degree but the response is less in 100 m/s wind speed also.Structure is not even failing but practical it is not possible right.And one question do you know about fragility of wind turbine?

**Reply by Krishnaraj18** [reply]
I have one more doubt. I started simulation from 1 m/s, But in that velocity itself i am getting around 0.5 m out-plane deflection.But there will be not this much deflection i refered journal also  in normal operating mode what are the things i have to edit in openFast OC4 jacket platform.? What are things i have to check.?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

I would say that tower deflection is driven by the aerodynamic loads on the rotor and tower and the blade deflection is driven by the aerodynamic loads on the rotor and gravity, but there is not a 1-to-1 correlation because of directionality of the forces, the pitch angle of the blades, and the different blade stiffness in flap and edge.

I agree that for parked simulations it best to set `WakeMod` = 0 and `AFAeroMod` = 1 in AeroDyn.

I would say that in most cases of variable-pitch wind turbines, the blades are pitch to 90 degrees in any parked or idling condition, regardless of the wind speed.

Regarding fragility, see a similar question discussed on our forum: https://forums.nrel.gov/t/deflection-of-elastodyn/2233.

I'm not sure I understand the rest of your questions.

Best regards,

**Reply by Krishnaraj18** [reply]
In parked only i have to change WakeMod and AFAeroMod?what about the other changes i have to edit ?I found n one journal like in intact parked mode only changing the angle 90 degree.

**Reply by jjonkman** [reply]
It sounds like you already know how, but to fix the blade-pitch angles to 90degrees, you set `BlPitch`(1-3) = 90 in ElastoDyn and disable pitch control in ServoDyn (`PCMode` = 0).

"Parking" a rotor implies that the brake is applied and generator will not spin, which can be achieved by disabling the generator DOF in ElastoDyn (`GenDOF` = False).  But it is more common to idle a rotor in storm conditions, which can be achieved by keeping `GenDOF` = TRUE but not engaging the generator by setting `GenTiStr` = TRUE with `TimGenOn` > `TMax` in ServoDyn.

Best regards,

**Reply by Krishnaraj18** [reply]
How can I account corrosion in OpenFast?Is there any option in OpenFast that i can given corrosion rate? And what about PSI,I found an option in PSI.But my doubt is how can i account PSI for a particular location? That i have to find in other software?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

Corrosion is not directly accounted for in OpenFAST.  Modeling corrosive damage would imply a change in structural properties (e.g. mass and/or stiffness) over time. This damage would not occur instantaneously such that changes in mass and/or stiffness would need to be considered over the course of a given simulation, which is typically 10 minutes or so in length. Instead, the effect of mass and/or stiffness changes can be realized by setting up different input files characteristic of each damaged state to be considered.

By "PSI", are you referring to local stress?  OpenFAST does not directly output stress; rather, OpenFAST outputs reaction loads at cross sections of flexible members, including shear and axial forces and bending and torsion moments. Computing stress or strain would require post-processing in a sectional analysis tool or structural FEA tool.

Best regards,

**Comment by Krishnaraj18** [comment]
Hii jjonkman,
I want to do soil structure effect on the response of the blade,tower and all the response of OC4 jacket wind turbine.I have to design the soil layer and planning to give distributed spring.My doubt is how can do it in openFast.In subDyn i found a SSI input file provision.But how to do it for distributed spring model for flexible foundation.Can you help me to figure it out.

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

SubDyn itself supports an apparent fixity (AF) representation of the foundation, as well as a coupled springs (CS) representation of the foundation; with the later available through the `SSIfile`.  But a distributed springs (DS) model is not currently supported.

The new SoilDyn module of OpenFAST (to be released soon through the following OpenFAST pull request: https://github.com/OpenFAST/openfast/pull/986) has a placeholder for future implementation of a distributed springs representation of the foundation. 

Best regards,

**Reply by Krishnaraj18** [reply]
Thank you,
i have one more doubt,in some journal they mentioned about FounDyn module to consider SSI.Where is that module actually in OpenFast?How it's works?

**Reply by Krishnaraj18** [reply]
If i am considering Coupled spring model,What should be the changes i have to do in OpenFast modules?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

I recall seeing a paper where FounDyn was mentioned, but this module was neither developed by, nor shared with, NREL.

To model a coupled springs (CS) foundation in SubDyn, set the reaction flags (`RctTDXss` through `RctRDZss`) to zero for each DOF of each reaction node you want stiffness to be considered, and set the stiffness matrix of the CS foundation via `SSIfile`.  See the online SubDyn documentation for more information:
https://openfast.readthedocs.io/en/main/source/user/subdyn/input_files.html#base-reaction-joints
https://openfast.readthedocs.io/en/main/source/user/subdyn/input_files.html#ssi-input-file

Best regards,

**Comment by Krishnaraj18** [comment]
Dear Jjonkman,
I have one doubt regarding SSI file in OpenFast.For considering pile soil interaction i am planning to give stiffness matrix as SSI file .But my doubt is how can i generate stiffness matrx,and which stiffness i have derive,pile head stiffness or what?And while giving SSI file what are the changes i have to do in dat files.And is there any chances of increasing the responses in blade ,tower and substructure?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

The SubDyn documentation I linked above explains the input file settings to define a coupled springs (CS) model of the foundation within SubDyn.  As far as how to derive the stiffness, in a CS model, the entire stiffness of the pile + soil within the foundation is lumped and represented as a stiffness matrix at the mudline at the point where the pile pierces the soil.  NREL has not supplied software to compute this CS stiffness, but you can read related papers on CS methods to understand the process.

Best regards,

**Comment by Krishnaraj18** [comment]
Dear Jjonkman,
I have on doubt regarding coupled spring model.In my project for considering Pile soil interaction, i used coupled spring model.I given SSI stiffness matrix as given 
4.96133E+07		 Kxx
0.00000E+00		 Kxy
4.94332E+07		 Kyy
0.00000E+00 		 Kxz
0.00000E+00 		 Kyz
1.40810E+08		 Kzz
0.00000E+00 		 Kxtx
1.24989E+07 		 Kytx
0.00000E+00 		 Kztx
2.93747E+10		 Ktxtx
-1.24989E+07		 Kxty
0.00000E+00 		 Kyty
0.00000E+00		 Kzty
0.00000E+00		 Ktxty
2.93747E+10		 Ktyty
0.00000E+00 		 Kxtz
0.00000E+00		 Kytz
0.00000E+00 		 Kztz
0.00000E+00 		 Ktxtz
0.00000E+00 		 Ktytz
7.56098E+08		 Ktztz
my soil type is sand,Is this values are reliable or not?And in the result i found no changes in blade deflection.But found a variation in tower top deflection and transition piece displacement.The transition piece displacement i found a large increase,but while looking to the Mudline bending moment,(BMx,BMy,BMz) i found no variation.Is it possible ?Why it's coming like this?Can you explain it.

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

I can't comment on the specific values you are using for the coupled spring (CS) stiffness matrix.  These will depend on many factors.

I'm not too surprised that adding soil stiffness will lead to increased motion of the tower-top and transition piece displacement.  I would guess that the external wind and wave excitation, which dominate the mudline loads, would not be much effected by the increased displacement, but I would think the increased displacement would have some effect on, e.g., the impact of RNA weight on the mudline loads.

Best regards,

**Comment by Krishnaraj18** [comment]
Dear Jjonkman,
I have done the dynamic simulation of the OC4 jacket wind turbine by providing the stiffness matrix mentioned in the previous discussion as an SSI.txt file in the Subdyn module. I found significant differences in the response of the tower and substructure. However, I want to find the new natural frequency of the entire structure, including the substructure, tower, and blades, due to the soil-structure interaction. How can I get the natural frequency?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

You can calculate full-system natural frequencies of OpenFAST models, including those with SubDyn enabled, through an OpenFAST linearization analysis, followed by eigenanalysis, as has been discussed in many topics here on OpenFAST GitHub, as well as on our forum (https://forums.nrel.gov/).

Best regards,
